test/cmd/error.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import "fmt"
+
+// CommandLineError is a custom error we use for errors got from running commands
+type CommandLineError struct {
```

Comment:
I assume this is used for returning `error`, how easy is it to use the fields like `ErrorCode` etc.? Do we need to cast the error to `CommandlineError` every time using this?

Reply:
Yes we'll need to cast the error to `CommandlineError` in order to get its fields. But I believe `ErrorCode` will not be used quite often, so it should be fine, and this is also how `exec.ExitError` works. I'm not directly using `exec.ExitError` since I want to add some other fields in `CommandlineError`.